File: generator/src/test/resources/kotlin/expected/ResponseEventsTest/test-event-source-method.output.kt
package io.test.service

import io.outfoxx.sunday.EventSource
import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import kotlin.collections.List

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(),
) {
  public suspend fun fetchEvents(): EventSource = this.requestFactory
    .eventSource(
      method = Method.Get,
      pathTemplate = "/tests",
      acceptTypes = listOf(MediaType.EventStream)
    )
}

File: generator/src/test/resources/swift/expected/ResponseEventsTest/test-event-source-method.output.swift
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = []
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchEvents() -> EventSource {
    return self.requestFactory.eventSource(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: [.eventStream],
      headers: nil
    )}

}
